roles/sideload_kernel/tasks/main.yml

```diff
@@ -0,0 +1,117 @@
+---
+- name: Validate required variables
+  ansible.builtin.assert:
+    that:
+      - sideload_kernel_uri | string
+
+- name: Install python dependencies
+  ansible.builtin.pip:
+    name:
+      - kubernetes
+
+- name: Sideload-kernel job namespace
+  kubernetes.core.k8s:
+    kubeconfig: "{{ kubeconfig }}"
+    state: present
+    api_version: v1
+    kind: Namespace
+    name: "{{ sideload_kernel_namespace }}"
+
+- name: Sideload-kernel configmap
+  kubernetes.core.k8s:
+    kubeconfig: "{{ kubeconfig }}"
+    state: present
+    apply: true
+    definition:
+      apiVersion: v1
+      kind: ConfigMap
+      metadata:
+        name: sideload-kernel
+        namespace: "{{ sideload_kernel_namespace }}"
+      data:
+        KERNEL_URI: "{{ sideload_kernel_uri }}"
+        flip_kernel: "{{ lookup('file', 'flip_kernel') }}"
+  register: configmap_created
+
+- name: Clean out any previous jobs
+  kubernetes.core.k8s:
+    kubeconfig: "{{ kubeconfig }}"
+    state: absent
+    api_version: batch/v1
+    kind: Job
+    name: flip-kernel
+    namespace: "{{ sideload_kernel_namespace }}"
+  when: configmap_created.changed or sideload_kernel_force | bool
+
+- name: Sideload-kernel job
+  kubernetes.core.k8s:
+    kubeconfig: "{{ kubeconfig }}"
+    state: present
+    definition:
+      apiVersion: batch/v1
+      kind: Job
+      metadata:
+        name: flip-kernel
+        namespace: "{{ sideload_kernel_namespace }}"
+      spec:
+        backoffLimit: 3
+        # TODO: Add a selector for multi-node cluster cases?
+        template:
+          spec:
+            containers:
+              - command:
+                  - /bin/sh
```

Comment:
did you test this with `/bin/sh`? I think I saw a few BASH-isms in the script

Reply:
oh I see, it does call `/tmp/flip_kernel` at end so it ends up running through whatever shebang it has

Reply:
But maybe using bash here too will remove any confusion :)